moz_preprocess/sample.sql

```diff
@@ -0,0 +1,9 @@
+--Append this SQL if downsampling clients is desired.
+
+)
+SELECT
+  * EXCEPT(clients_per_segment)
+FROM
+  clients_weekly
```

Comment:
I will admit the SQL is a bit overwhelming, so please disregard if this isn't relevant. But I wonder if we could add this to the base query so that you don't need to do string matching/replacing on line 47 in `bq_utils.py` -- that strategy can be brittle if someone deletes the comment, adds a newline or space before `SELECT`, etc.

I'm wondering if we could change `weekly_aggregate.sql` to something like this instead:
```
WITH
  search_clients AS (
  [... more SQL code ...]
  clients_weekly AS (
  SELECT
    COUNT(DISTINCT client_id) OVER (PARTITION BY segment) as clients_per_segment,
    [... more SQL code ...]
  )
SELECT
  * EXCEPT(clients_per_segment)
FROM
  clients_weekly
{where_clause}
```

and then we add the filter and other string replacements in `bq_utils.py` like:

```
if sample:
    where_clause = f"WHERE RAND() < {sample} / clients_per_segment"
else:
    where_clause = ""

sql = sql.format(
    week_start_date = week_start_date,
    segment = segment,
    target = target,
    where_clause = where_clause,
)
```

I'm not sure if the SQL I'm proposing would work in the same way that you have here, but if so this approach would enable us to simplify the sampling/string replacement code and remove this file.

Reply:
I love this idea